src/Sentry/Internal/Hub.cs

```diff
@@ -154,7 +154,9 @@ public async Task ConfigureScopeAsync(Func<Scope, Task> configureScope)
                 transaction.SampleRate = sampleRate;
             }
 
-            if (transaction.IsSampled is true && _options.TransactionProfilerFactory is { } profilerFactory)
+            if (transaction.IsSampled is true &&
+                _options.TransactionProfilerFactory is { } profilerFactory &&
+                _randomValuesFactory.NextBool(_options.ProfilesSampleRate ?? 0.0))
```

Comment:
Can we avoid the null coalescing -> random generation if we don't have a sample rate? 
Might even be able to do something like `_options.ProfilesSampleRate is { } sampleRate && sampleRate > 0.0
```suggestion
            if (transaction.IsSampled is true &&
                _options.TransactionProfilerFactory is { } profilerFactory &&
                _options.ProfilesSampleRate is { } sampleRate &&
                sampleRate > 0.0 &&
                _randomValuesFactory.NextBool(sampleRate))
```

Reply:
I've had it like that but:
1. NextBool short-circuits if the given value is 0.0
2. it actually never is because options.TransactionProfilerFactory is null if profiling is disabled (as checked by the iOS & .net profiler integrations